omod-1.11/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_11/OrderableResource1_11.java

```diff
@@ -0,0 +1,198 @@
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ * {@link org.openmrs.module.webservices.rest.web.annotation.Resource} for
+ * {@link org.openmrs.ConceptSearchResult}, this resource is used to search for concepts that are
+ * orderable. It returns ConceptSearchResults instead of Concepts
+ */
+@Resource(name = RestConstants.VERSION_1 + "/orderable", supportedClass = Concept.class, supportedOpenmrsVersions = {
+        "1.11.*", "1.12.*", "2.0.*" })
+public class OrderableResource1_11 extends BaseDelegatingResource<ConceptSearchResult> implements Searchable {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#newDelegate()
+	 */
+	@Override
+	public ConceptSearchResult newDelegate() {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#save(Object)
+	 */
+	@Override
+	public ConceptSearchResult save(ConceptSearchResult delegate) {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	@Override
+	public SimpleObject search(RequestContext context) throws ResponseException {
+		
+		String query = context.getParameter("q");
+		
+		List<OrderType> orderTypes = Context.getOrderService().getOrderTypes(false);
+		List<ConceptClass> orderableConceptClasses = getOrderableConceptClasses(orderTypes);
+		
+		List<ConceptClass> conceptClasses = null;
+		String[] classUuids = context.getRequest().getParameterValues("conceptClasses");
+		
+		if (classUuids != null) {
+			for (String uuid : classUuids) {
+				ConceptClass cc = getConceptClassByUuid(uuid, orderableConceptClasses);
+				if (cc != null) {
+					if (conceptClasses == null) {
+						conceptClasses = new ArrayList<ConceptClass>();
+					}
+					
+					conceptClasses.add(cc);
+				}
+			}
+		}
+		
+		String[] orderTypeUuids = context.getRequest().getParameterValues("orderTypes");
+		if (orderTypeUuids != null) {
+			
+			for (String uuid : orderTypeUuids) {
+				// look for order type in the orders collections
+				OrderType orderType = getOrderTypeByUuid(uuid, orderTypes);
+				
+				if (orderType != null) {
+					
+					Collection<ConceptClass> orderTypeConceptClasses = orderType.getConceptClasses();
+					if (orderableConceptClasses != null) {
+						
+						if (conceptClasses == null) {
+							conceptClasses = new ArrayList<ConceptClass>();
+						}
+						
+						for (ConceptClass cc : orderTypeConceptClasses) {
+							
+							if (!conceptClasses.contains(cc))
+								conceptClasses.add(cc);
+						}
+					}
+				}
+			}
+		}
+		
+		if (conceptClasses == null | (conceptClasses != null && conceptClasses.size() == 0))
```

Comment:
also if the first case (`conceptClasses == null`) is not true you dont need to check again that its not null.

so this is enough

`if (conceptClasses == null || conceptClasses.size() == 0)`

and I think you should use this

`if (conceptClasses == null || conceptClasses.isEmpty())`

since this shows your intent more clearly and isEmpty might also be faster than computing the size and checking if that is 0